File: java/Sets/setOperations.java
/*
In this problem, we will implement the set operations of inter-
section and union. The class SetUtilities contains the starter code for
this problem. You may not use the retainAll() or addAll() methods of
a set in this problem.
*/


import java.util.Set;
import java.io.FileNotFoundException;
import java.util.HashSet;

// implement union and intersection
public class setOperations {
    public static void main(String[] args) {
        Set<Integer> set1 = new HashSet<Integer>(); 
        Set<Integer> set2 = new HashSet<Integer>();
        set1.add(3);
        set1.add(2);
        set1.add(4);
        set1.add(1);


        set2.add(5);
        set2.add(3);
        set2.add(99);
        set2.add(1);
        Set<Integer> result = findIntersection(set1, set2);
        Set<Integer> result2 = findUnion(set1, set2);
        // System.out.println(result);  
        
        

    }

    private static Set findUnion(Set<Integer> set1, Set<Integer> set2) {
        Set<Integer> resultSet = new HashSet();
        resultSet.addAll(set1);
        resultSet.addAll(set2);
        return resultSet;
    }

    private static Set findIntersection(Set<Integer> set1, Set<Integer> set2) {

        // we only need to convert one set into an array
        Integer[] set1Arr = set1.toArray(new Integer[set1.size()]);
        Set<Integer> resultSet = new HashSet<Integer>();

        // if any items in our converted array is in set2 we add it to our result set
        for(int i = 0; i < set1Arr.length; i++){
            if(set2.contains(set1Array[i])){
                resultSet.add(set1Arr[i]);
            }
        }
        return resultSet;

    }
}

File: swift/Sets/setOperations.swift
import Foundation

class SetOperations {

    // O(1) -
    static func findUnion(setObj set1: Set<Int>, setObj set2: Set<Int>) -> Set<Int> {
        // works similar to just throwing both set into an array
        // then iterating over that array to see if our result set contains any elements in the array
        return set1.union(set2)
    }
}

let set1: Set<Int> = [3, 2, 4, 1]
let set2: Set<Int> = [5, 3, 99, 1]
let result = SetOperations.findUnion(setObj: set1, setObj: set2)
print(result)
